src/overlays/actors/ovl_Bg_Heavy_Block/z_bg_heavy_block.c

```diff
@@ -21,35 +38,497 @@ const ActorInit Bg_Heavy_Block_InitVars = {
+    ICHAIN_VEC3F(scale, 1, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 400, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 400, ICHAIN_STOP),
+};
+
+extern UNK_TYPE D_0600169C;
+extern Gfx D_060013C0[];
+extern Gfx D_06001A30[];
+extern Gfx D_060018A0[];
+
+void BgHeavyBlock_SetPieceRandRot(BgHeavyBlock* this, f32 scale) {
+    this->dyna.actor.posRot.rot.x = Math_Rand_CenteredFloat(1024.0f) * scale;
+    this->dyna.actor.posRot.rot.y = Math_Rand_CenteredFloat(1024.0f) * scale;
+    this->dyna.actor.posRot.rot.z = Math_Rand_CenteredFloat(1024.0f) * scale;
+}
+
+void BgHeavyBlock_InitPiece(BgHeavyBlock* this, f32 scale) {
+    f32 rand;
+    f32 yawSinCos;
+    f32 randChoice;
+
+    this->dyna.actor.gravity = -0.6f;
+    this->dyna.actor.minVelocityY = -12.0f;
+    randChoice = Math_Rand_CenteredFloat(12.0f * scale);
+    rand = (randChoice < 0.0f) ? randChoice - 2.0f : randChoice + 2.0f;
+    this->dyna.actor.velocity.y = (Math_Rand_ZeroFloat(8.0f) + 4.0f) * scale;
+    this->dyna.actor.velocity.z = Math_Rand_ZeroFloat(-8.0f * scale);
+    yawSinCos = Math_Coss(this->dyna.actor.posRot.rot.y);
+    this->dyna.actor.velocity.x =
+        (Math_Sins(this->dyna.actor.posRot.rot.y) * this->dyna.actor.velocity.z + (yawSinCos * rand));
+    yawSinCos = Math_Sins(this->dyna.actor.posRot.rot.y);
+    this->dyna.actor.velocity.z =
+        (Math_Coss(this->dyna.actor.posRot.rot.y) * this->dyna.actor.velocity.z) + (-yawSinCos * rand);
+    BgHeavyBlock_SetPieceRandRot(this, scale);
+    Actor_SetScale(&this->dyna.actor, Math_Rand_CenteredFloat(0.2f) + 1.0f);
+}
+
+void BgHeavyBlock_SetupDynapoly(BgHeavyBlock* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    UNK_TYPE a1;
+
+    a1 = 0;
+    this->dyna.actor.flags |= 0x20030;
+    DynaPolyInfo_SetActorMove(&this->dyna, 0);
+    DynaPolyInfo_Alloc(&D_0600169C, &a1);
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, a1);
+}
+
+void BgHeavyBlock_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgHeavyBlock* this = THIS;
+
+    Actor_ProcessInitChain(thisx, sInitChain);
+    ActorShape_Init(&thisx->shape, 0.0f, NULL, 0.0f);
+    this->pieceFlags = 0;
+
+    if (globalCtx->sceneNum == SCENE_GANON_TOU) {
+        thisx->params &= 0xFF00;
+        thisx->params |= 4;
+    }
+
+    switch (thisx->params & 0xFF) {
+        case 2: // bigger broken piece
+            thisx->draw = BgHeavyBlock_DrawPiece;
+            this->actionFunc = BgHeavyBlock_MovePiece;
+            BgHeavyBlock_InitPiece(this, 1.0f);
+            this->timer = 120;
+            thisx->flags |= 0x10;
+            this->unk_164.y = -50.0f;
+            break;
+        case 3: // smaller broken piece
+            thisx->draw = BgHeavyBlock_DrawPiece;
+            this->actionFunc = BgHeavyBlock_MovePiece;
+            BgHeavyBlock_InitPiece(this, 2.0f);
+            this->timer = 0x78;
```

Comment:
Looks like I missed this before
```suggestion
            this->timer = 120;
```